### PR TITLE
PARQUET-1196: Example parquet_arrow project

### PR DESCRIPTION
**Pull request by xhochy**
Depends on https://github.com/apache/parquet-cpp/pull/434

### CODE REVIEW

**Comment by xhochy** [comment]
@lorenzhs another one for you!

File: cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -48,10 +48,6 @@ if (DEFINED ENV{THRIFT_HOME})
   set(THRIFT_HOME "$ENV{THRIFT_HOME}")
 endif()
 
-if (DEFINED ENV{ARROW_HOME})
-  set(ARROW_HOME "$ENV{ARROW_HOME}")
-endif()
```
Comment (wesm) [inline cmake_modules/ThirdpartyToolchain.cmake]: Moving this to FindArrow.cmake was surprising, but I'm not sure if there's a better way

File: examples/parquet-arrow/src/reader-writer.cc [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  std::shared_ptr<arrow::io::ReadableFile> infile;
+  PARQUET_THROW_NOT_OK(arrow::io::ReadableFile::Open(
+      "parquet-arrow-example.parquet", arrow::default_memory_pool(), &infile));
+
+  std::unique_ptr<parquet::arrow::FileReader> reader;
+  PARQUET_THROW_NOT_OK(
+      parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
+  std::shared_ptr<arrow::Array> array;
+  PARQUET_THROW_NOT_OK(reader->ReadColumn(0, &array));
+  arrow::PrettyPrint(*array, 4, &std::cout);
```
Comment (wesm) [inline examples/parquet-arrow/src/reader-writer.cc]: Check status?

File: examples/parquet-arrow/src/reader-writer.cc [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  std::shared_ptr<arrow::io::ReadableFile> infile;
+  PARQUET_THROW_NOT_OK(arrow::io::ReadableFile::Open(
+      "parquet-arrow-example.parquet", arrow::default_memory_pool(), &infile));
+
+  std::unique_ptr<parquet::arrow::FileReader> reader;
+  PARQUET_THROW_NOT_OK(
+      parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
+  std::shared_ptr<arrow::Array> array;
+  PARQUET_THROW_NOT_OK(reader->RowGroup(0)->Column(0)->Read(&array));
+  arrow::PrettyPrint(*array, 4, &std::cout);
```
Comment (wesm) [inline examples/parquet-arrow/src/reader-writer.cc]: Status

**Pull request closed without merging**